psl-parser/src/main/java/org/linqs/psl/parser/CommandLineLoader.java

```diff
@@ -0,0 +1,421 @@
+//delete below 4 imports? never used in both files?
+/* 
+import org.apache.log4j.BasicConfigurator;
+import org.apache.log4j.ConsoleAppender;
+import org.apache.log4j.PatternLayout;
+import org.apache.log4j.Priority;
+*/
+import org.apache.log4j.PropertyConfigurator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.net.UnknownHostException;
+import java.net.InetAddress;
+import java.nio.file.Paths;
+import java.util.Comparator;
+import java.util.Properties;
+import java.util.Map;
+
+/**
+ * Load the command line options into PSL Config's configuration values and 
+ * log4j configuration appropriately. This is done by instatiating the object
+ * of type CommandLine and object log of type Logger.
+ */
+public class CommandLineLoader {
+    // Command line options.
+    public static final String OPTION_HELP = "h";
+    public static final String OPTION_HELP_LONG = "help";
+    public static final String OPERATION_INFER = "i";
+    public static final String OPERATION_INFER_LONG = "infer";
+    public static final String OPERATION_LEARN = "l";
+    public static final String OPERATION_LEARN_LONG = "learn";
+
+    public static final String OPTION_DATA = "d";
+    public static final String OPTION_DATA_LONG = "data";
+    public static final String OPTION_DB_H2_PATH = "h2path";
+    public static final String OPTION_DB_POSTGRESQL_NAME = "postgres";
+    public static final String OPTION_EVAL = "e";
+    public static final String OPTION_EVAL_LONG = "eval";
+    public static final String OPTION_INT_IDS = "int";
+    public static final String OPTION_INT_IDS_LONG = "int-ids";
+    public static final String OPTION_LOG4J = "4j";
+    public static final String OPTION_LOG4J_LONG = "log4j";
+    public static final String OPTION_MODEL = "m";
+    public static final String OPTION_MODEL_LONG = "model";
+    public static final String OPTION_OUTPUT_DIR = "o";
+    public static final String OPTION_OUTPUT_DIR_LONG = "output";
+    public static final String OPTION_OUTPUT_GROUND_RULES_LONG = "groundrules";
+    public static final String OPTION_OUTPUT_SATISFACTION_LONG = "satisfaction";
+    public static final String OPTION_PROPERTIES = "D";
+    public static final String OPTION_PROPERTIES_FILE = "p";
+    public static final String OPTION_PROPERTIES_FILE_LONG = "properties";
+    public static final String OPTION_VERSION = "v";
+    public static final String OPTION_VERSION_LONG = "version";
+
+    public static final String DEFAULT_H2_DB_PATH =
+    Paths.get(System.getProperty("java.io.tmpdir"),
+    "cli_" + System.getProperty("user.name") + "@" + getHostname()).toString();
+    public static final String DEFAULT_POSTGRES_DB_NAME = "psl_cli";
+    public static final String DEFAULT_IA = MPEInference.class.getName();
+    public static final String DEFAULT_WLA = MaxLikelihoodMPE.class.getName();
+
+    public CommandLine options;
```

Comment:
it was so in Launcher.java. moved it directly here.. changed DEFAULT_IA and DEFAULT_WLA from public to private.

Reply:
No.
Constants should, in general, be public and member datum private.
We may want to reference `DEFAULT_WLA` from elsewhere (as we do with many constants), and since it is static and final that is not an issue.